src/history/test/HistoryTests.cpp

```diff
@@ -320,6 +346,100 @@ TEST_CASE("History publish to multiple archives", "[history]")
     REQUIRE(catchupSimulation.catchupOffline(catchupApp, checkpointLedger));
 }
 
+TEST_CASE("Publish works correctly post shadow removal", "[history]")
```

Comment:
In general this test and the next one look like they're testing some pretty subtle conditions and it's difficult for me to extract what you're aiming to check or how. A little commentary would help here about the overall test plan and what is essentially under investigation vs. merely incidental to the test.

(They look like good tests! I just .. don't quite understand what they're all about)

Reply:
Ah great, I can follow it much more directly now. The commentary makes it clear how the cases differ and why you're testing them, thanks!

Reply:
fixed